### PR TITLE
vstart: make -k with optional mon_num.

### PR DESCRIPTION
**Pull request by majianpeng**
Default mon_num=3, if we change this and using -k to create a new
cluster. It don't work. So we must record the old mon_num.
The same for osd_num/mds_num/rgw_num.

Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

### CODE REVIEW

File: src/vstart.sh [hunk 1]
```diff
@@ -275,6 +275,13 @@ esac
 shift
 done
 
+if [ "$overwrite_conf" -eq 0 ]; then
+  CEPH_NUM_MON=`cat $conf_fn|grep CEPH_NUM_MON|awk -F= '{print $2}'`
```
Comment (tchaikov) [inline src/vstart.sh]: nit, could just put
```awk
CEPH_NUM_MON=`awk -F= '/CEPH_NUM_MON/{print $2}' $conf_fn`
```

**Comment by tchaikov** [comment]
lgtm apart from a nit.

**Comment by dillaman** [comment]
This PR now causes `MON=1 OSD=1 MDS=0 ./vstart.sh -k` to fail:

`./vstart.sh: line 428: [: 1: unary operator expected`

File: src/vstart.sh [hunk 2]
```diff
@@ -275,6 +275,13 @@ esac
 shift
 done
 
+if [ "$overwrite_conf" -eq 0 ]; then
+  CEPH_NUM_MON=`awk -F= '/CEPH_NUM_MON/{print $2}' $conf_fn`
```
Comment (dillaman) [inline src/vstart.sh]: Can we default these if the `awk` fails?

Comment (tchaikov) [inline src/vstart.sh]: ahh, i overlooked this. will fix it tomorrow.

**Pull request merged**